Fix for location merge issue

mo-nathan opened the pull request:
This fixes the issue described in #2630, but this is just a quick fix which I don't think is the approach we really want to take.

mo-nathan committed:
Hack to fix the immediate issue

mo-nathan committed:
Test for expect link

nimmolo commented on app/helpers/tabs/observations_helper.rb:
```diff
@@ -145,14 +145,20 @@ def observations_at_where_tabs(query)
+    def where_param(params)
+      params[:user_where] || params[:locations][0]
+    end
+
```
It looks like Igor's example page (that has the `define_location_tab`) had the **incoming** `where` param you need:
`https://mushroomobserver.org/observations?where=Eberwhite+Nature+Area%2C+Washtenaw+Co.%2C+Michigan%2C+USA`

Does `params[:user_where] || params[:where]` not work? IIRC `user_where` would only come from an Advanced search.

nimmolo replied:
Oh, i notice you're using this below too.

nimmolo replied:
OH. I get it. There's a confusion here between `query.params` and ACTUAL params. Hang on, i'm going to commit what I believe is a fix.

nimmolo replied:
Fix seems to work.

nimmolo commented on app/helpers/tabs/observations_helper.rb:
```diff
@@ -145,14 +145,20 @@ def observations_at_where_tabs(query)
 
+    # Hack to use the :locations param if it's present and the :user_where
+    # param is missing.
+    def where_param(params)
+      params[:user_where] || params[:locations][0]
+    end
```
```suggestion
    def where_param(query_params)
      query_params[:user_where] || params[:where]
    end
```

nimmolo committed:
Update observations_helper.rb

A value for `query.params[:user_where]` would only come from a previously saved `AdvancedSearch` query.

What we want to handle here is the Rails `params[:where]` that sent us to this page.

mo-nathan commented:
The coveralls failures aren't real.
